### PR TITLE
Fix bintray/upload and add tests.

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**

### CODE REVIEW

File: bin/bintray/upload [hunk 1]
```diff
@@ -138,6 +148,7 @@ do
 		r) comp_release="$OPTARG" ;;
 		p) comp_prerelease="$OPTARG" ;;
 		a) arch="$OPTARG" ;;
+		n) dry_run=true ;;
```
Comment (joseph-wakeling-sociomantic) [inline bin/bintray/upload:151]: .... how many times did you re-read this line just in case? :-P

File: test/bintray/upload/1 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+BINTRAY_USER=test TRAVIS_TAG=v1.2.3 TRAVIS_REPO_SLUG=sociomantic/test beaver bintray upload -N -C test -k test -n lala.deb
+0
+jfrog bt version-create --desc= --vcs-tag=v1.2.3 --released= sociomantic/test/test/v1.2.3
```
Comment (joseph-wakeling-sociomantic) [inline test/bintray/upload/1]: Is the space after `--released` intentional here?

File: test/bintray/test [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+set -u
+
+stdout=$(mktemp beaver.test.stdout.XXXXXXXX)
+stderr=$(mktemp beaver.test.stderr.XXXXXXXX)
+expout=$(mktemp beaver.test.expout.XXXXXXXX)
```
Comment (joseph-wakeling-sociomantic) [inline test/bintray/test:6]: Can you explain these lines?  I'm not sure I follow the intention.

File: bin/bintray/upload (hunk 1 above)
Comment (leandro-lucarella-sociomantic) [reply bin/bintray/upload:151]: Hopefully only once now that we have tests :wink:

File: test/bintray/upload/1 (hunk 2 above)
Comment (leandro-lucarella-sociomantic) [reply test/bintray/upload/1]: Not sure, I mean, `--released` should be a date, and it's empty. Not sure if it should be empty, I will check.

File: test/bintray/test (hunk 3 above)
Comment (leandro-lucarella-sociomantic) [reply test/bintray/test:6]: I'm just creating temporary files to be used by the tests...

File: test/bintray/upload/1 (hunk 2 above)
Comment (leandro-lucarella-sociomantic) [reply test/bintray/upload/1]: Yeah, this is OKish, but not very clean. The released date is taken from the Git tag date, but in the fabricated tests there is no Git tag, hence it is empty. Not ideal but fixing that could be challenging (we basically should have some face git repo to get the info from).

Comment (leandro-lucarella-sociomantic) [reply test/bintray/upload/1]: Same goes for `--desc=`.

**Comment by leandro-lucarella-sociomantic** [comment]
OK, green.

**Comment by leandro-lucarella-sociomantic** [comment]
I'll merge after I get any approvals and then I release `rc.2`.

File: bin/bintray/upload [hunk 4]
```diff
@@ -114,17 +118,23 @@ bt()
...
+	if ! $dry_run
+	then
+		# Dummy config to avoid jfrog to stop and ask questions...
+		jfrog bt config --user=nobody --key=nokey --licenses=nolicense
+	fi
 	echo jfrog bt "$cmd" "$@"
-	jfrog bt "$cmd" --user="$user" --key="$key" "$@"
+	if ! $dry_run
+	then
+		jfrog bt "$cmd" --user="$user" --key="$key" "$@"
```
Comment (joseph-wakeling-sociomantic) [inline bin/bintray/upload:129]: Just a minor note: this line appears different from what is `echo`'d above.  Is that intentional?

Comment (joseph-wakeling-sociomantic) [reply bin/bintray/upload:129]: (I recognize the line is unchanged from the previous version, but now that we're actually using an official dry-run mode, having the command accurate might be more important.)

**Review by joseph-wakeling-sociomantic — APPROVED** [review]
LGTM.  Let's get that RC on the road ;-)

**Comment by leandro-lucarella-sociomantic** [comment]
Out!

**Comment by joseph-wakeling-sociomantic** [comment]
This now works -- see MXNet package builds:

* rc: https://travis-ci.org/sociomantic-tsunami/mxnet/builds/390741333

* stable: https://travis-ci.org/sociomantic-tsunami/mxnet/builds/390755097

... so I think you can proceed to final release.

**Pull request merged**